### PR TITLE
Remove seperation between ArcGis Feature Service and ArcGIS Map Service items in browser

### PR DESCRIPTION
**Pull request by nyalldawson**
Builds off https://github.com/qgis/QGIS/pull/40697:

Replaces the separate "ArcGis Feature Service" and "ArcGIS Map Service" browser items/connections with a single, unified "ArcGIS Rest Servers" connection and browser node which shows a definitive view of the server exposing both service types.

This makes it MUCH easier to manage and browse these servers, and matches user instead of the awkward forced separation between the layer types.

### CODE REVIEW

**Comment by nirvn** [comment]
I'm a big +1 on this move. Makes a lot of sense and de-clutters our browser panel as well as data source dialog.

**Comment by nyalldawson** [comment]
I've also added other improvements here:

1. There's now a unified "ArcGIS REST Server" tab in the Data Source Manager, which shows the combined feature service/map service views of servers
2. The Data Source Manager widget is now based off browser models, so it's nice and responsive and loads in background threads, and only loads data on demand. Previously we walked the ENTIRE server in one blocking call on the main thread, which would lead to 10 min+ ui hangs on large servers. Ouch!
3. Removed the never-hooked-up, non functional query buttons for feature service layers
4. Made the search line edit actually work

File: src/providers/arcgisrest/qgsarcgisrestdataitemguiprovider.cpp [hunk 1]
```diff
@@ -142,21 +160,21 @@ void QgsAfsDataItemGuiProvider::deleteConnection( QgsDataItem *item )
...
 
-void QgsAfsDataItemGuiProvider::saveConnections()
+void QgsArcGisRestDataItemGuiProvider::saveConnections()
 {
   QgsManageConnectionsDialog dlg( nullptr, QgsManageConnectionsDialog::Export, QgsManageConnectionsDialog::ArcgisFeatureServer );
   dlg.exec();
 }
 
-void QgsAfsDataItemGuiProvider::loadConnections( QgsDataItem *item )
+void QgsArcGisRestDataItemGuiProvider::loadConnections( QgsDataItem *item )
```
Comment (nirvn) [inline src/providers/arcgisrest/qgsarcgisrestdataitemguiprovider.cpp:177]: Do we need to add code to handle pre 3.18 saved connections that'd include ArcgisMapServer?

File: src/providers/arcgisrest/qgsarcgisrestdataitems.cpp [hunk 2]
```diff
@@ -0,0 +1,674 @@
...
+
+QString QgsArcGisRestConnectionItem::url() const
+{
+  const QgsOwsConnection connection( QStringLiteral( "ARCGISFEATURESERVER" ), mConnName );
+  return connection.uri().param( QStringLiteral( "url" ) );
+}
+
+
+//
+// QgsAfsGroupsItem
```
Comment (nirvn) [inline src/providers/arcgisrest/qgsarcgisrestdataitems.cpp:238]: There seems to be a bunch of these section comments that are out of sync with the subsequent class name, e.g. QgsAfsGroupsItem vs QgsArcGisPortalGroupsItem.

File: src/providers/arcgisrest/qgsarcgisrestdataitemguiprovider.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/providers/arcgisrest/qgsarcgisrestdataitemguiprovider.cpp:177]: That's done here: https://github.com/qgis/QGIS/pull/40700/files#diff-5fab4f0ef4a0231dcfa1488222c365a5955f1df79cfaf06e4086275ea6bd542bR611

File: src/providers/arcgisrest/qgsarcgisrestdataitems.cpp (hunk 2 above)
Comment (nyalldawson) [reply src/providers/arcgisrest/qgsarcgisrestdataitems.cpp]: Fixed, thanks

**Pull request merged**